### PR TITLE
[WIP] Added changeInBalance to updatedUserBalance Event

### PR DESCRIPTION
**Pull request by mayorcoded**
This pr is a feature that adds a changeInBalance variable to the updatedUserBalance event as described in the 
issue: https://github.com/MARKETProtocol/MARKETProtocol/issues/152. The pr does the following
- add changeInBalance paramter to UpdatedUserBalance event.
- calculates change in balance by doing changeInBalance = newBalance - initialBalance in functions where 
  updatedUserBalance is emitted.
- emit new UpdatedUserBalance event with changeInBalance argument.

Tests:
- functions `depositTokensForTrading` and `withdrawTokens` were tested.
- private functions `commitCollateralToPool` and `withdrawCollateralFromPool` were not tested (any ideas?)
- test result: 93 pass, 1 fail.
- The failing test is from the MarketContractOraclize contract. And from my investigation, this test fails when
  the code branch is not the develop branch. The test does not fail in the develop branch, but fails in any 
  branch at all whether the code base changes or not.

CI checks:
93.65% Stmts|76.88% Branch|94.17% Funcs|92.81% Lines

### CODE REVIEW

File: contracts/MarketCollateralPool.sol [hunk 1]
```diff
@@ -112,9 +112,11 @@ contract MarketCollateralPool is Ownable {
     /// @param depositAmount qty of ERC20 tokens to deposit to the smart contract to cover open orders and collateral
     function depositTokensForTrading(address collateralTokenAddress, uint256 depositAmount) external {
         uint256 balanceAfterDeposit = tokenAddressToAccountBalance[collateralTokenAddress][msg.sender].add(depositAmount);
+        uint256 balanceBeforeDeposit = tokenAddressToAccountBalance[collateralTokenAddress][msg.sender];
+        int256 changeInBalance = int256(balanceAfterDeposit).subtract(int256(balanceBeforeDeposit));
```
Comment (pelsasser) [inline contracts/MarketCollateralPool.sol]: Shouldn't this amount be the exact same amount as `depositAmount` represented as an integer?

File: contracts/MarketCollateralPool.sol [hunk 2]
```diff
@@ -190,10 +192,11 @@ contract MarketCollateralPool is Ownable {
         uint256 balanceAfterWithdrawal = tokenAddressToAccountBalance[collateralTokenAddress][msg.sender].subtract(
             withdrawAmount
         );
-
+        uint256 balanceBeforeWithdrawal = tokenAddressToAccountBalance[collateralTokenAddress][msg.sender];
+        int256 changeInBalance = int256(balanceAfterWithdrawal).subtract(int256(balanceBeforeWithdrawal));
```
Comment (pelsasser) [inline contracts/MarketCollateralPool.sol]: same thing here, shouldn't it always be equal to the withdraw amount?

File: contracts/MarketCollateralPool.sol (hunk 1 above)
Comment (mayorcoded) [reply contracts/MarketCollateralPool.sol]: Yes, @pelsasser it should be the same! I already mentioned that in the comments I made while explaining my solution before I started coding, I just did not get a clear direction about it from your replies.

File: contracts/MarketCollateralPool.sol (hunk 2 above)
Comment (mayorcoded) [reply contracts/MarketCollateralPool.sol]: Yes, @pelsasser it should be the same! I already mentioned that in the comments I made while explaining my solution before I started coding, I just did not get a clear direction about it from your replies.

**Comment by mayorcoded** [comment]
Thanks, @pelsasser for the heads-up about the architecture change coming up. I will be willing to help resolve issues in the new architecture when it is available. I appreciate very much the comments you gave me. It was very encouraging for me.

**Comment by pelsasser** [comment]
:) Great! Closing this PR and the associated issue for now.

**Pull request closed without merging**